src/OpenSpaceToolkit/Astrodynamics/Conjunction/Messages/CCSDS/CDM.cpp

```diff
@@ -0,0 +1,747 @@
+    object1CovarianceMatrix(2, 0) = std::stod(aDictionary["SAT1_CN_R"].accessString()) ;
+    object1CovarianceMatrix(2, 1) = std::stod(aDictionary["SAT1_CN_T"].accessString()) ;
+    object1CovarianceMatrix(2, 2) = std::stod(aDictionary["SAT1_CN_N"].accessString()) ;
+    object1CovarianceMatrix(3, 0) = std::stod(aDictionary["SAT1_CRDOT_R"].accessString()) ;
+    object1CovarianceMatrix(3, 1) = std::stod(aDictionary["SAT1_CRDOT_T"].accessString()) ;
+    object1CovarianceMatrix(3, 2) = std::stod(aDictionary["SAT1_CRDOT_N"].accessString()) ;
+    object1CovarianceMatrix(3, 3) = std::stod(aDictionary["SAT1_CRDOT_RDOT"].accessString()) ;
+    object1CovarianceMatrix(4, 0) = std::stod(aDictionary["SAT1_CTDOT_R"].accessString()) ;
+    object1CovarianceMatrix(4, 1) = std::stod(aDictionary["SAT1_CTDOT_T"].accessString()) ;
+    object1CovarianceMatrix(4, 2) = std::stod(aDictionary["SAT1_CTDOT_N"].accessString()) ;
+    object1CovarianceMatrix(4, 3) = std::stod(aDictionary["SAT1_CTDOT_RDOT"].accessString()) ;
+    object1CovarianceMatrix(4, 4) = std::stod(aDictionary["SAT1_CTDOT_TDOT"].accessString()) ;
+    object1CovarianceMatrix(5, 0) = std::stod(aDictionary["SAT1_CNDOT_R"].accessString()) ;
+    object1CovarianceMatrix(5, 1) = std::stod(aDictionary["SAT1_CNDOT_T"].accessString()) ;
+    object1CovarianceMatrix(5, 2) = std::stod(aDictionary["SAT1_CNDOT_N"].accessString()) ;
+    object1CovarianceMatrix(5, 3) = std::stod(aDictionary["SAT1_CNDOT_RDOT"].accessString()) ;
+    object1CovarianceMatrix(5, 4) = std::stod(aDictionary["SAT1_CNDOT_TDOT"].accessString()) ;
+    object1CovarianceMatrix(5, 5) = std::stod(aDictionary["SAT1_CNDOT_NDOT"].accessString()) ;
+
+    for (int i = 0; i < 5; i++)
+    {
+        for (int j = 1; j < 6; j++)
+        {
+            object1CovarianceMatrix(i, j) = object1CovarianceMatrix(j, i) ;
+        }
+    }
+
+    CDM::Data sat1ObjectData = CDM::Data
+    {
+        Instant::Undefined(),
+        Instant::Undefined(),
+        Duration::Undefined(),
+        Duration::Undefined(),
+        Integer::Undefined(),
+        Integer::Undefined(),
+        Integer::Undefined(),
+        Integer::Undefined(),
+        Real::Undefined(),
+        Real::Undefined(),
+        Real::Undefined(),
+        Real::Undefined(),
+        Real::Undefined(),
+        Mass::Undefined(),
+        Real::Undefined(),
+        Real::Undefined(),
+        Real::Undefined(),
+        Real::Undefined(),
+        State(
+            Instant::DateTime(DateTime::Parse(aDictionary["TCA"].getString(), DateTime::Format::ISO8601), Scale::UTC),
+            Position::Meters(  // TBI: Add Position::Kilometers
+                {
+                    Real::Parse(aDictionary["SAT1_X"].accessString()) * 1000.0,
+                    Real::Parse(aDictionary["SAT1_Y"].accessString()) * 1000.0,
+                    Real::Parse(aDictionary["SAT1_Z"].accessString()) * 1000.0
+                },
+                Frame::ITRF()
```

Comment:
Do you confirm this is the right frame to use here? (Since another frame seems used for another Position field)

Reply:
So it is not necessarily `ITRF` in the general case indeed. What we should do is check based on the value of the other `refFrame` field.

Reply:
Yeah let's, and just fail at loading time if the frame is not something OSTK can support as of today.

Reply:
My plan is to do a separate pass on those Frame/Position/Velocity/Covariance fields (might need to add other Frame functions) and adapt where needed. I'll experiment with the other fields for now

Reply:
Right, my only ask is to not have implicit hardcoding because we may/will forget to revisit, and this can lead to subtle bugs. So if we only support ITRF for now, totally fine, but let's blatantly fail if the CDM is expressed in a different frame.

Reply:
Yep okay, that makes total sense. I'll modify this :)

Reply:
Adapted to support `ITRF`, `GCRF` and `J2000` based on options enumerated on standard at https://public.ccsds.org/Pubs/508x0b1e2s.pdf.